libcxx/test/std/ranges/range.factories/range.iota.view/empty.pass.cpp

```diff
@@ -0,0 +1,136 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17
+
+// constexpr bool empty() const;
+
+#include <cassert>
+#include <concepts>
+#include <ranges>
+#include <utility>
+#include <vector>
+
+#include "types.h"
+
+template <typename R>
+concept HasFreeEmpty = requires(R r) { std::ranges::empty(r); };
+
+template <typename R>
+concept HasMemberEmpty = requires(R r) {
+  { r.empty() } -> std::same_as<bool>;
+};
+
+constexpr void test_empty_iota() {
+  std::vector<int> ev;
+
+  // Both parameters are non-const
+  {
+    auto iv = std::views::iota(std::ranges::begin(ev), std::ranges::end(ev));
+
+    static_assert(HasFreeEmpty<decltype(iv)>);
+    static_assert(HasMemberEmpty<decltype(iv)>);
+
+    assert(iv.empty());
+  }
+  // Left parameter is const
+  {
+    auto iv = std::views::iota(std::ranges::begin(std::as_const(ev)), std::ranges::end(ev));
+
+    static_assert(HasFreeEmpty<decltype(iv)>);
+    static_assert(HasMemberEmpty<decltype(iv)>);
+
+    assert(iv.empty());
+  }
+  // Right parameter is const
+  {
+    auto iv = std::views::iota(std::ranges::begin(ev), std::ranges::end(std::as_const(ev)));
```

Comment:
Question: does using `as_const` really matter to test `iv.empty()`?

Reply:
I thought it was good to be detailed. Should I remove those?

Reply:
Removed